src/tests/unit/tests/electron/platform/android/scan-results-helpers.ts

```diff
@@ -208,27 +207,67 @@ function buildAtfaSpannableString(rawString: string): SpannableString {
     return s as SpannableString;
 }
 
-function buildAtfaElement(
-    accessibilityClassName: string,
-    id: number,
-    className: string,
-    boundsInScreen?: any,
-    contentDescription?: string,
-    text?: string,
-): ViewHierarchyElement {
+function buildAtfaSpannableStringViaSpan(textViaSpan: string): SpannableString {
+    const s = {};
+    s['SpannableString.rawString'] = { 'SpannableStringInternal.mText': textViaSpan };
+    return s as SpannableString;
+}
+
+function verifyAndBuildSpannableString(
+    fieldName: string,
+    value?: string,
+    valueViaSpan?: string,
+): SpannableString | null {
+    if (value && valueViaSpan) {
+        throw Error(`${fieldName} can't be both direct and indirect!`);
+    }
+
+    if (value) {
+        return buildAtfaSpannableString(value);
+    }
+
+    if (valueViaSpan) {
+        return buildAtfaSpannableStringViaSpan(valueViaSpan);
+    }
+
+    return null;
+}
+
+function buildAtfaElement(elementParameters: AtfaElementParameters): ViewHierarchyElement {
     const e = {};
-    e['ViewHierarchyElement.accessibilityClassName'] = accessibilityClassName;
-    e['ViewHierarchyElement.id'] = id;
-    e['ViewHierarchyElement.className'] = className;
+    e['ViewHierarchyElement.accessibilityClassName'] = elementParameters.accessibilityClassName;
+    e['ViewHierarchyElement.className'] = elementParameters.className;
+
+    if (elementParameters.boundsInScreen) {
+        e['ViewHierarchyElement.boundsInScreen'] = buildAtfaRectangle(
+            elementParameters.boundsInScreen,
+        );
+    }
 
-    if (boundsInScreen) {
-        e['ViewHierarchyElement.boundsInScreen'] = buildAtfaRectangle(boundsInScreen);
+    if (elementParameters.contentDescription && elementParameters.contentDescriptionViaSpan) {
+        throw Error("contentDescription can't be both direct and indirect");
     }
+    if (elementParameters.text && elementParameters.textViaSpan) {
+        throw Error("text can't be both direct and indirect");
+    }
```

Comment:
Are these lines needed if `verifyAndBuildSpannableString` throws the same error message? It looks like `verifyAndBuildSpannableString` is called for both the `contentDescription` and `text` below.

Reply:
Oops! I refactored and forgot to remove these. Will fix